src/main/java/minefantasy/mfr/block/BlockForge.java

```diff
@@ -172,38 +172,48 @@ public boolean isBurning(IBlockAccess world, BlockPos pos) {
 	@Override
 	public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, EntityPlayer player, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
 		ItemStack held = player.getHeldItemMainhand();
-		TileEntityForge tile = (TileEntityForge) getTile(world, pos);
-		if (tile != null) {
-			if (tile.getIsLit() && !ItemApron.isUserProtected(player)) {
+		TileEntityForge forge = (TileEntityForge) getTile(world, pos);
+		if (forge != null) {
+			/// Burn unprotected players
+			if (forge.getIsLit() && !ItemApron.isUserProtected(player)) {
 				player.setFire(5);
 				player.attackEntityFrom(DamageSource.ON_FIRE, player.isWet() ? 3 : 1);
 				if (!player.world.isRemote) {
 					player.sendMessage(new TextComponentTranslation("info.noHeatProtection.message"));
 				}
 			}
 			if (!held.isEmpty()) {
-				if (facing == EnumFacing.UP && held.getItem() instanceof ItemTongs && onUsedTongs(player, held, tile)) {
+				/// Tong use
+				if (facing == EnumFacing.UP && held.getItem() instanceof ItemTongs && onUsedTongs(player, held, forge)) {
 					return true;
 				}
-				int uses = ItemLighter.tryUse(held, player);
-				if (!tile.getIsLit() && uses != 0) {
-					player.playSound(SoundEvents.ITEM_FLINTANDSTEEL_USE, 1.0F, 1.0F);
-					if (uses == 1) {
-						tile.fireUpForge();
-						held.damageItem(1, player);
+				/// Ignition
+				if (held.getItem() instanceof ItemFlintAndSteel || held.getItem() instanceof ILighter) {
+					if (!forge.getIsLit() && forge.getFuel() > 0 && forge.getTier() != 1) {
+						int uses = ItemLighter.tryUse(held, player);
+						if (uses != 0) // 1 for ignition, -1 for a failed attempt, 0 for a null input or for an item that needs to bypass normal ignition
```

Comment:
Oh - that's something I started doing in my own code to denote labels rather than just snippets that have been commented out. However, this isn't my codebase, so I understand why it feels out of place (especially given that one of my goals was to standardize things). I'll switch it to adhere to normal convention